Phoenix/Core/Source/Voxels/Blocks.cpp

```diff
@@ -38,28 +36,26 @@ RegisteredBlock::RegisteredBlock(const std::string& unique, int id,
 
 RegisteredBlock::~RegisteredBlock() = default;
 
-BlockRegistry::BlockRegistry() : i(0) {};
-
-BlockRegistry::~BlockRegistry() {};
+BlockRegistry::BlockRegistry() : m_BlockIdCounter(0) {};
 
 int BlockRegistry::registerBlock(const std::string& uniqueName,
                                  const std::string& displayName)
 {
-	i++;
-	Blocks[i] = RegisteredBlock(uniqueName, i, displayName);
-	return i;
+	m_BlockIdCounter++;
+	m_Blocks.push_back(RegisteredBlock(uniqueName, m_BlockIdCounter, displayName));
+	return m_BlockIdCounter;
 };
 
 const std::string& BlockRegistry::getDisplayName(int blockId)
 {
-	return Blocks[blockId].displayName;
+	return m_Blocks[blockId].displayName;
 };
 
 int BlockRegistry::getBlockId(const std::string& uniqueName)
 {
-	for (i = 0; i < 100; i++)
+	for (int i = 0; i < 100; i++)
 	{
-		if (Blocks[i].uniqueName == uniqueName)
+		if (m_Blocks[i].uniqueName == uniqueName)
 		{
 			return i;
```

Comment:
So this returns the 0-based index as the ID - should it not return the ID stored inside `m_blocks[i]` instead of the index - due to (as mentioned in a previous comment) the stored ID's are 1-indexed, but this will return the (0-indexed) index of the element and not it's (1-indexed) ID.